gemini/use-cases/retrieval-augmented-generation/llamaindex_rag.ipynb

```diff
@@ -0,0 +1,1437 @@
+        "!gcloud config set project {PROJECT_ID}\n",
+        "# !gcloud auth application-default login"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e48fbda5ff18"
+      },
+      "source": [
+        "### Restart runtime\n",
+        "\n",
+        "To use the newly installed packages in this Jupyter runtime, you must restart the runtime. You can do this by running the cell below, which restarts the current kernel.\n",
+        "\n",
+        "The restart might take a minute or longer. After it's restarted, continue to the next step."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7a918c599327"
+      },
+      "outputs": [],
+      "source": [
+        "import IPython\n",
+        "\n",
+        "app = IPython.Application.instance()\n",
+        "app.kernel.do_shutdown(True)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bef4a1021f6f"
+      },
+      "source": [
+        "### Authenticate your notebook environment (Colab only)\n",
+        "\n",
+        "If you're running this notebook on Google Colab, run the cell below to authenticate your environment."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f460f33d2158"
+      },
+      "outputs": [],
+      "source": [
+        "import sys\n",
+        "\n",
+        "if \"google.colab\" in sys.modules:\n",
+        "    from google.colab import auth\n",
+        "\n",
+        "    auth.authenticate_user()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4x10VEywwN3Z"
+      },
+      "source": [
+        "### Using LlamaIndex"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-2mDLUbdweJi"
+      },
+      "source": [
+        "![LlamaRAG](https://storage.googleapis.com/github-repo/generative-ai/gemini/use-cases/retrieval-augmented-generation/llamaindex/LlamaRAG.png)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ybyxRabnmzlI"
+      },
+      "source": [
+        "### Download sample data\n",
+        "Refer to document 04a02.pdf\n",
+        "\n",
+        "This document describes the importance of stable power grids in Japan, highlighting the recent failure of a generator step-up transformer at the Nakoso Power Station and the rapid restoration response undertaken to maintain power supply stability.\n",
+        "\n",
+        "We will use this pdf moving forward."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0f2f9cac2beb"
+      },
+      "outputs": [],
+      "source": [
+        "!mkdir ./data\n",
+        "!gcloud storage cp gs://github-repo/generative-ai/gemini/use-cases/retrieval-augmented-generation/llamaindex/data/* ./data\n",
```

Comment:
change this to use the `DOC_FOLDER` variable from earlier.

https://nocomplexity.com/documents/jupyterlab/tip-passingshellvariables.html

Reply:
Fixed!